### PR TITLE
[TC-346] MKR WiFi 1010 App Note

### PR DESCRIPTION
**Pull request by aliphys**
## What This PR Changes
- Add application note for connecting a LiPo battery to the MKR boards (MKR WiFi 1010)

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

File: content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+### Electrical Properties
+
+**Voltage**
+The nominal voltage of both LiPo and Li-Ion batteries is around 3.7V and is commonly referred to as the voltage of single cell batteries. However, this is an ideal scenario since as a battery is used the voltage changes. The voltage can be considered similar to a waterfall. The higher the waterfall, the higher the voltage.
+
+
+In the MKR boards, the battery terminal is connected to the SAMD21 via a reserved pin (PB09) known as `ADC_BATTERY` within the Arduino Core. This pin is used internally on the board, and is not accessible via the MKR pins. Since the voltage of a Li-ion battery exceeds 3.3V (the AREF value), a voltage divider must be used to extend the range while also ensure that only safe voltages are applied to the microcontroller. We can calculate the output voltage using the following formula
+
+$$ V_{out} = \frac{V_{source} \times R2} {R_1 + R_2} $$
```
Comment (karlsoderby) [inline content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md:67]: Hey @aliphys. Nicely written document for the MKR board. But I think I discovered a little bug here, which is the use of "$" as a styling component? It doesn't render anything at docs if you take a look at it, so I am not sure how you can render it differently? Like using `backticks` maybe?

Comment (aliphys) [reply content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md:67]: Thanks for catching the render issue. The `$` sign is used to render math in Tex. It renders OK locally.

@jhansson-ard What syntax do we use for Tex in the `docs-content` repo?

**Comment by karlsoderby** [comment]
Hey @aliphys is there anything needed to go live with this article?

**Comment by aliphys** [comment]
> Hey @aliphys is there anything needed to go live with this article?

Other people have also reported issues with MathJax and Gatsby e.g. https://github.com/gatsbyjs/gatsby/issues/16983
For this task, I'll render images, and upload them instead.

File: content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
+---
+title: 'Understanding Battery Capabilities Of The MKR 1010 WiFi'
+difficulty: intermediate
+description: 'Choose the correct battery for you MKR application.'
```
Comment (karlsoderby) [inline content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md:4]:
```suggestion
description: 'Learn about different topics related to batteries, and how to choose the correct battery for you MKR application.'

```

File: content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
+---
+title: 'Understanding Battery Capabilities Of The MKR 1010 WiFi'
```
Comment (karlsoderby) [inline content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md:2]: I think this summarizes the article better.

```suggestion
title: 'MKR WiFi 1010 Battery Application Note'

```

File: content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+  - Arduino IDE 2.x
+---
+
+## Introduction
+
+While you can run your MKR board from USB, you can make best use of it when your MKR board is able to operate independently of your computer and a USB cable. This is particularly useful for many IoT applications, use of relays with high current draw and remote sensing. Batteries can help us in this regard and a special connector is included in the MKR products for this end. In this application note, we will take a closer look at the battery capabilities of the MKR boards. 
+
+### Goals
+The goals of this project are to:
+- Understand what the terms `charging capacity` and `charging current`
```
Comment (karlsoderby) [inline content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md:22]: Use backticks only for code. When highlighting something like this, let's bold them.

```suggestion
- Understand what the terms **charging capacity** and **charging current** mean.

```

File: content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+- Distinguish between LiPo and Li-Ion batteries
+- Learn how to identify connector polarity
+- Understand how the SAMD chip reads battery voltage via a voltage divider
+
+### Hardware & Software Needed
+- [Arduino MKR WiFi 1010](https://store.arduino.cc/products/arduino-mkr-wifi-1010)
+- Multimeter
+- IDE 2.x
```
Comment (karlsoderby) [inline content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md:30]:
```suggestion
- [Arduino IDE](https://www.arduino.cc/en/software)

```

**Review by karlsoderby — CHANGES_REQUESTED** [review]
Looks good @aliphys . Left some comments that needs to be addressed before merging.

File: content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+**17.** Finally, we will add a half a second delay at the end of the `loop()` function to allow variables to come slowly through the Serial Monitor.
+```arduino
+  delay(500);
+}
+```
+
+**18.** You section for the `loop()` function should look as follows.
+![Battery level calculation and Serial printing](assets/calculation-serial-printing.png)
+
+**11.** The complete code (with the addition of comments) is as follows. You can copy and paste this directly into your IDE
```
[the remark this answers is not in the document]
Comment (aliphys) [reply content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md:224]:
```suggestion
**19.** The complete code (with the addition of comments) is as follows. You can copy and paste this directly into your IDE

```

File: content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md (hunk 4 above)
Comment (aliphys) [reply content/hardware/01.mkr/01.boards/mkr-wifi-1010/tutorials/mkr-battery-app-note/mkr-battery-app-note.md:22]: Noted!

**Comment by karlsoderby** [comment]
Hey @aliphys you will need to rebase this PR because of a gatsby issue we had last week. Also there are some small issues to fix with the **spellcheck** and **content linter** checks.

**Comment by aliphys** [comment]
@karlsoderby So I should rebase it again?

**Comment by karlsoderby** [comment]
@aliphys Ah I see that you did that now! I saw that gatsby fail and I assumed it was due to that issue. I re-ran the check now so hopefully it works :)

**Comment by aliphys** [comment]
Current PR branch superseeds the previous (https://github.com/arduino/docs-content/tree/aliphys/mkrBatteryAppNote ) branch, hence it is deleted

**Comment by aliphys** [comment]
@karlsoderby can we merge this app noe now? :)

**Comment by aliphys** [comment]
Gatsby building now working, re-based branch and now up to date 🎉

**Review by marqdevx — APPROVED** [review]
Approving, I did not review

**Pull request merged**